### PR TITLE
FEATURE: [atrpin] take profit by expected base balance

### PR DESCRIPTION
**Pull request by narumiruna**

### CODE REVIEW

File: pkg/strategy/atrpin/strategy.go [hunk 1]
```diff
@@ -133,25 +154,34 @@ func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, se
...
+			s.logger.Infof("position: %+v", position)
+			baseQuantity = position.Base
+		}
+
+		side := types.SideTypeSell
+		takerPrice := ticker.Buy
+		if baseQuantity.Sign() < 0 {
+			side = types.SideTypeBuy
+			takerPrice = ticker.Sell
+			baseQuantity = baseQuantity.Abs()
```
Comment (c9s) [inline pkg/strategy/atrpin/strategy.go:169]: I think you can check the ClosePosition() method, and maybe we can reuse some of them

Comment (narumiruna) [reply pkg/strategy/atrpin/strategy.go:169]: If there is a missing trade, then the position base will be incorrect. Thus I'd like to add the `ExpectedBaseBalance` to solve the problem.

Comment (c9s) [reply pkg/strategy/atrpin/strategy.go]: can we use query trades to fix the position?

the balance balance could be fixed by xalign, and we will need a correct PnL record

Comment (narumiruna) [reply pkg/strategy/atrpin/strategy.go]: we can't fix the balance by xalign, because atrpin will place order

Comment (narumiruna) [reply pkg/strategy/atrpin/strategy.go]: how to query trades?

**Review by c9s — APPROVED** [review]

**Pull request merged**